src/embedded.ts

```diff
@@ -523,3 +545,9 @@ export class RoomWidgetClient extends MatrixClient {
         }
     }
 }
+
+function processAndThrow(error: unknown): never {
+    throw error instanceof WidgetApiResponseError && error.data.matrix_api_error
+        ? MatrixError.fromWidgetApiErrorData(error.data.matrix_api_error)
+        : error;
```

Comment:
this is unreadable, can we have an if statement with 2 throws instead please?

Reply:
b6f8d1412